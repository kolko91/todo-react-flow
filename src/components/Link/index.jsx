import React, { Component, type Node } from 'react';
import styled from 'styled-components';


export type Props = {
  active: boolean,
  children?: Node,
  onClick: () => void
};
const StyledLink = styled.a`
    color: inherit;
    margin: 3px;
    padding: 3px 7px;
    text-decoration: none;
    border: 1px solid transparent;
    border-radius: 3px;
    &:hover{
      border-color: rgba(175, 47, 47, 0.1);
      cursor: pointer;
    }
`;

const Link = ({ active, children, onClick }: Props) => {
  if (active) {
    return <span>{children}</span>;
  }

  return (
    <StyledLink
      href="#"
      onClick={(event: Event) => {
        event.preventDefault();
        onClick();
      }}
    >
      {children}
    </StyledLink>
  );
};
Link.defaultProps = {
  children: '',
};

export default Link;

