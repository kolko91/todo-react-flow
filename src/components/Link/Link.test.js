import React from 'react';
import renderer from 'react-test-renderer';
import Link from './index';


describe('component', () => {
  describe('Link', () => {
    it('should render correctly', () => {
      const props = {
        active: false,
        children: 'All',
        onClick: jest.fn(),
      };
      const tree = renderer.create(<Link {...props} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
