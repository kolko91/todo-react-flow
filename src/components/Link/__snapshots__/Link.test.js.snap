// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component Link should render correctly 1`] = `
<a
  className="sc-bdVaJa dKuIrI"
  href="#"
  onClick={[Function]}
>
  All
</a>
`;
